### ISSUE TITLE
How to make peace with needed nested import statements for Ruff E402?

### ISSUE THREAD

**Issue by alanwilter** [astral-sh/ruff#9091]
I use this lines a lot in several projects.

```python
import matplotlib
import numpy as np

matplotlib.use("Agg")

from matplotlib import pyplot as plt
from matplotlib.backends.backend_pdf import PdfPages
```

There's no other way that I know to set the matplotlib backend. But by doing so this will trigger [E402](https://docs.astral.sh/ruff/rules/module-import-not-at-top-of-file/).

`isort` has some ways of handling that (see [here](https://pycqa.github.io/isort/docs/configuration/action_comments.html)).

Of course I can fill lots of `# noqa: E402` but this is so ugly.

Ideally, Ruff/isort should be smart enough to know these cases.

Also, perhaps, page [E402](https://docs.astral.sh/ruff/rules/module-import-not-at-top-of-file/) (and all others when reasonable) should show how to silence this warning if the user knows what's doing.

**Comment by charliermarsh** [comment]
I've run into this `matplotlib` pattern before. I think it's actually reasonable for us to allow `matplotlib.use` calls to intersperse imports... In practice, I suspect it will strictly reduce false positives, with no false negatives.

**Comment by charliermarsh** [comment]
I'll PR it and see what the ecosystem checks say.

**Comment by alanwilter** [comment]
You guys are amazing. Father Xmas won't forget you ;-)

**Comment by district10** [comment]
So what's the solution?

I have a similar problem, `matplotlib.use("Agg")` -> `load_dotenv()`.
I don't want to <*fill lots of `# noqa: E402`*> (this is so ugly).

**Comment by ntBre** [comment]
Looking at the linked PR, `matplotlib.use` was added as a special case, so I don't think there's a similar fix for `load_dotenv`. We do support the mentioned [isort](https://pycqa.github.io/isort/docs/configuration/action_comments.html) action comments in [Ruff](https://docs.astral.sh/ruff/linter/#action-comments) now, so that's one option. Otherwise it might make sense to open a separate, more general issue for discussion.
